Command/GenerateRoutes.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace HandcraftedInTheAlps\RestRoutingBundle\Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Yaml\Yaml;
+
+class GenerateRoutes extends Command
+{
+    private string $projectDirectory;
+    private RouterInterface $router;
+
+    public function __construct(string $projectDirectory, RouterInterface $router)
+    {
+        $this->projectDirectory = $projectDirectory;
+        $this->router = $router;
+
+        parent::__construct('fos-routing:generate-symfony');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $routes = [];
+        foreach ($this->router->getRouteCollection() as $name => $route) {
+            $routeData = $route->__serialize();
+            // Unset things that are probably defaulted by Symfony
+            unset($routeData['options']['compiler_class']);
+            unset($routeData['options']['utf8']);
+            $routeData['controller'] = $routeData['defaults']['_controller'] ?? '';
+            unset($routeData['defaults']['_controller']);
+
```

Comment:
just to keep the order of the params more logical:

```suggestion
            $defaults= $routeData['defaults'] ?? [];
            $requirements= $routeData['requirements'] ?? [];
            unset($routeData['defaults']);
            unset($routeData['requirements']);
            if ($defaults !== []) {
                 $routeData['defaults'] = $defaults;
            }
            if ($requirements !== []) {
                 $routeData['requirements'] = $requirements;
            }

```